extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/integration/QuarkusClassFileLocator.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.hibernate.orm.deployment.integration;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import net.bytebuddy.dynamic.ClassFileLocator;
+import net.bytebuddy.utility.StreamDrainer;
+
+public final class QuarkusClassFileLocator implements ClassFileLocator {
```

Comment:
Maybe a short explanation of why this class is better than the default impl would help? Let's think of the poor souls having to handle bugfixes after this gets merged :D

Reply:
ah right. Yes in fairness this isn't better than the standard `ClassFileLocator` yet, as it's exactly the same actually except not needing the complicated custom classloader to defer to.
But I also have plans to make it much better in a subsequent commits so to get rid of the expansion logic that creates multiple `byte[]` - IFF it all works: I'd like to know if it's worth doing that first.